controllers/DataSetAdmin.js

```diff
@@ -0,0 +1,271 @@
+  }
+
+  /**
+   * Save datarows uploaded from CSV,
+   * associating each datarow with the parent dataset
+   * and gathering summary field information about the collection of rows
+   * to store in the parent.
+   * 
+   */
+  dataSetUploadDataRowSave(req, res) {
+
+    let setId = req.param('id')
+    this.log.debug( "dataSetUploadDataRowSave setId ", setId, " req ", _.keys(req))
+    if(!req.file) {
+      req.flash('error', "Error processing rows: no file supplied for the data.")
+      return res.redirect(this.routePrefix)
+    }
+    /*
+      TODO: integrate the dataManager and define/initialize
+     */
+    return dataManager.importFileToModel('datasets-datarow', req.file.path, {type: 'csv', truncate: true, strict: false})
+    .then((rows) => {
+      let currentFields = []
+      _.each(rows, (rowElem, index) => {
+        this.log.debug("saveDataUpload " + index + " processing rowElem", rowElem)
+        if (Array.isArray(rowElem)) {
+          rowElem = rowElem[0] //work-around for model handler dupe bug in data-loader 3.0.0 (do we need this in 4.0??)
+        }
+        if (_.isEmpty(currentFields)) {
+          currentFields = this._buildFieldInfo(rowElem)
+        }
+        if (setId) {
+          let rowToUpdate = this._prepareDataRowInDataSet(rowElem, setId)
+          rowToUpdate.save().catch((err) => {
+            this.log.error("nxus-dataset update failed for datarow " + rowElem.id, err)
+          })
+        }
+      })
+      return this.model.findOne(setId).then((set) => {
+        //warn about limitation of current implementation if set already has rows.
+        if ( 0 < set.rowCount ) {
+          this.log.error("Adding additional rows to a dataset isn't supported yet")
+          req.flash('info', "Uable to add data rows to set with data.")
+          return res.redirect(this.routePrefix)
+        } 
+        set.rowCount = rows.length
+        set.fields = currentFields
+        this.log.debug( "nxus-dataset saveDataUpload dataset " + setId + " fields:", currentFields )
+        return set.save().then( () => {
+          req.flash('info', "Successfully loaded " + rows.length + " data rows")
+          res.redirect(this.routePrefix) //TODO -- route to the edit form for this dataset instead?
+        })
+      })
+    }).catch((e) => {
+      this.log.error( "nxus-dataset saveDataUpload() error loading data rows  ", e, " at: ", e.stack);
+      req.flash('error', "Error processing rows")
+      return res.redirect(this.routePrefix)
+    })
+  }
+
+  /**
+   * Transform the datarow after storage,
+   * associating with the given dataset ID.
+   * @param  {[datarow]} row       DataRow instance to put under given DataSet ID
+   * @param  {[string]} dataSetId   Parent DataSet ID
+   * @return the original row passed in, with modifications.
+   */
+  _prepareDataRowInDataSet(row, dataSetId) {
+    _.each(row, (val,key) => {
+      if (typeof(val) == 'string') {
+        if (val.endsWith('%')) {

```

Comment:
Would ideally like to consolidate into a shared or existing library for this type validation and parsing